pyccel/codegen/printing/ccode.py

```diff
@@ -1092,7 +1092,10 @@ def _cast_to(self, expr, dtype, precision):
 
     def _print_DottedVariable(self, expr):
         """convert dotted Variable to their C equivalent"""
-        return '{}.{}'.format(self._print(expr.lhs), self._print(expr.name))
+        if self.stored_in_c_pointer(expr):
+            return '(*{}.{})'.format(self._print(expr.lhs), self._print(expr.name))
```

Comment:
Why not `{}->{}`?
Won't the `*` be added in `self._print(expr.lhs)`?

Reply:
i don't remember adding this code

Reply:
No this is right. It is `expr` that is stored in a c pointer not `expr.lhs`